File: main/src/model/Goal.java
package model;

import java.time.LocalDateTime;
import java.util.Date;

public class Goal {
    private final LocalDateTime makeDate;
    private final LocalDateTime deadline;

    public Goal() {
        LocalDateTime date = LocalDateTime.now();
        makeDate = date;
        deadline = date.plusMonths(1);
    }

    public LocalDateTime getMakeDate() {
        return makeDate;
    }

    public LocalDateTime getDeadline() {
        return deadline;
    }
}

File: main/src_ts/model/Goal.ts

export default class Goal {
    private _date: Date = new Date();
    private readonly _makeDate: Date;
    private readonly _deadline: Date;

    constructor() {
        this._makeDate = this._date;
        this._deadline = this._date;
        this._deadline.setMonth(this._deadline.getMonth() + 1);
    }

    get date(): Date {
        return this._date;
    }

    get makeDate(): Date {
        return this._makeDate;
    }

    get deadline(): Date {
        return this._deadline;
    }
}